### PR TITLE
8339698: x86 unused andw/orw/xorw/addw encoding could be removed

### PR DESCRIPTION
**Pull request by sviswa7**
x86 andw/orw/xorw(Register, Register) encoding is missing 0x66 prefix. These instructions along with addw(Register, Register) are unused and so removed.

Best Regards,
Sandhya

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339698](https://bugs.openjdk.org/browse/JDK-8339698): x86 unused andw/orw/xorw/addw encoding could be removed (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Jatin Bhateja](https://openjdk.org/census#jbhateja) (@jatin-bhateja - **Reviewer**)
 * [Quan Anh Mai](https://openjdk.org/census#qamai) (@merykitty - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20901/head:pull/20901` \
`$ git checkout pull/20901`

Update a local copy of the PR: \
`$ git checkout pull/20901` \
`$ git pull https://git.openjdk.org/jdk.git pull/20901/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20901`

View PR using the GUI difftool: \
`$ git pr show -t 20901`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20901.diff">https://git.openjdk.org/jdk/pull/20901.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20901#issuecomment-2335002389)

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by merykitty** [comment]
May I ask what are these instructions used for, and can we simply replace them with the dword version?

File: src/hotspot/cpu/x86/assembler_x86.cpp [hunk 1]
```diff
@@ -1633,6 +1633,7 @@ void Assembler::andb(Address dst, Register src) {
 }
 
 void Assembler::andw(Register dst, Register src) {
+  emit_int8(0x66);
```
Comment (jatin-bhateja) [inline src/hotspot/cpu/x86/assembler_x86.cpp:1636]: This is operand-size override prefix which modifies the default operand size16bits.

**Review by jatin-bhateja — APPROVED** [review]

**Comment by merykitty** [comment]
I mean what do we use these instructions for, I think that if these are not needed we can remove them instead.

**Comment by sviswa7** [comment]
I am ok either way, removing these instructions or fixing the encoding. @vnkozlov Please let me know your thoughts.

**Comment by vnkozlov** [comment]
> I am ok either way, removing these instructions or fixing the encoding. @vnkozlov Please let me know your thoughts.

Remove them. Check `addw(Register dst, Register src)` - I think it is not used too.

All these instructions were added for first JEP 8223347: Integration of Vector API (Incubator)
May be there were experiments with using them during development.

**Comment by sviswa7** [comment]
@vnkozlov @merykitty I have updated the PR implementing your review comments.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by sviswa7** [comment]
Thanks a lot @vnkozlov.

**Review by jatin-bhateja — APPROVED** [review]
Cleanup is also fine, still looks good.

**Review by merykitty — APPROVED** [review]

**Comment by sviswa7** [comment]
Thanks @jatin-bhateja @merykitty for the review.

**Pull request closed without merging**